src/settings/components/SettingsLayout.jsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { Outlet, useLocation } from 'react-router-dom';
+import PageLayout from '../../common/components/PageLayout';
+import SettingsMenu from './SettingsMenu';
+
+const commondBreadcrumb = 'settingsTitle';
+
+const routes = {
+  settings: {
+    announcement: 'serverAnnouncement',
+    calendars: 'sharedCalendars',
+    calendar: 'sharedCalendar',
+    commands: 'sharedSavedCommands',
+    command: 'sharedSavedCommand',
+    attributes: 'sharedComputedAttributes',
+    attribute: 'sharedComputedAttribute',
+    devices: 'deviceTitle',
+    device: 'sharedDevice',
+    drivers: 'sharedDrivers',
+    driver: 'sharedDriver',
+    geofence: 'sharedGeofence',
+    groups: 'settingsGroups',
+    group: 'groupDialog',
+    maintenances: 'sharedMaintenance',
+    maintenance: 'sharedMaintenance',
+    notifications: 'sharedNotifications',
+    notification: 'sharedNotification',
+    preferences: 'sharedPreferences',
+    server: 'settingsServer',
+    users: 'settingsUsers',
+    user: 'settingsUser',
+  },
+};
+
+const compoundRoutes = {
```

Comment:
This looks really complicated. Why do we need all of these? I think you should be able to split it by path segements.

Reply:
What do you think of this alternative?
https://gist.github.com/jorgeart81/6140d1afeab31ae690b11eedf8dca106

Reply:
Seems like it's still overcomplicated. Why can't we just append to an array? Something like:
```
let breadcrumbs = ['settingsTitle'];
if (length >= 2) {
  breadcrumbs.add(routes[pathSegmets[1]);
}
if (length >= 3) {
  switch ...
}
```

Reply:
I think that although it works for the current context, using [push ](https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Global_Objects/Array/push) would not be correct because push is used to add elements to an array, every time it is called.

I propose another solution so as not to affect the structure of your code too much. PR -> https://github.com/traccar/traccar-web/pull/1253

Reply:
The other PR looks promising, if it works as expected. Commented there.